### PR TITLE
Fixes runtime `radio.dm, line 267: list index out of bounds`

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Fixes runtime `radio.dm, line 267: list index out of bounds`

## Why It's Good For The Game
bug fix

## Testing Photographs and Procedure

I want to skip this

## Changelog
:cl:
fix: Fixes runtime `radio.dm, line 267: list index out of bounds`
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -261,15 +261,16 @@
...
-		if(channel == MODE_DEPARTMENT && channels.len > 0)
+	if(channel && length(channels) && length(secure_radio_connections))
+		if(channel == MODE_DEPARTMENT)
 			channel = channels[1]
 		freq = secure_radio_connections[channel]
 		if(istype(M, /mob) && !freq && channel != RADIO_CHANNEL_UPLINK)
 			to_chat(M, "<span class='warning'>You can't access this channel without an encryption key!</span>")
 		if (!channels[channel]) // if the channel is turned off, don't broadcast
 			return
-	else
```
Comment (Rukofamicom) [inline code/game/objects/items/devices/radio/radio.dm:272]: This seems like an arbitrary change though - a frequency is always set in the if statement above, this is a minor but unnecessary extra calculation every time this runs without the else.

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
Seems like a good fix to me

File: code/game/objects/items/devices/radio/radio.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/game/objects/items/devices/radio/radio.dm:272]: It's not really an arbitrary change. You can pass the if check while you can failed to get any value from `secure_radio_connections` list. That's why I made `if(!freq)`

**Comment by Rukofamicom** [comment]
Wait why is this closed? 

This was a good fix, I only wanted to address a single line.

**Comment by EvilDragonfiend** [comment]
> Wait why is this closed?
> 
> This was a good fix, I only wanted to address a single line.

Pigeon actually made a fix https://github.com/BeeStation/BeeStation-Hornet/pull/10362

**Pull request closed without merging**